koku/masu/database/trino_sql/reporting_awscostentrylineitem_summary_by_ec2_compute_p.sql

```diff
@@ -38,27 +38,54 @@ with cte_pg_enabled_keys as (
       from postgres.{{schema | sqlsafe}}.reporting_enabledtagkeys
      where enabled = true
      and provider_type = 'AWS'
+),
+cte_latest_values as (
+    SELECT
+        lineitem_resourceid as resource_id,
+        nullif(product_instancetype, '') as instance_type,
+        max(json_extract_scalar(json_parse(resourcetags), '$.Name')) AS instance_name,
+        resourcetags as tags,
+        costcategory as cost_category
+    FROM hive.{{schema | sqlsafe}}.aws_line_items_daily as alid
+    WHERE source = '{{source_uuid | sqlsafe}}'
+    AND year = '{{year | sqlsafe}}'
+    AND month = '{{month | sqlsafe}}'
+    AND lineitem_productcode = 'AmazonEC2'
+    AND product_productfamily LIKE '%Compute Instance%'
+    AND lineitem_resourceid != ''
+    AND lineitem_usagestartdate = (
+      SELECT max(date(lv.lineitem_usagestartdate)) AS usage_start
```

Comment:
So, right now we only grab latest values for the latest day, but have you tested if a VM was spun down partially through the report period. Say that the VM was spun down on the 9th, would we still get the latest values for the 9th?

Reply:
I think if we specify the source in the inner query, we can grab the latest from that specific VM.

Reply:
I think the change you added to have the resource_ids match helped a ton here too:
```
trino:org1234567> SELECT
               ->     lineitem_resourceid as resource_id,
               ->     nullif(product_instancetype, '') as instance_type,
               ->     max(json_extract_scalar(json_parse(resourcetags), '$.Name')) AS instance_name,
               ->     resourcetags as tags,
               ->     costcategory as cost_category
               -> FROM aws_line_items_daily as alid
               -> WHERE source = 'b52d9062-c53d-4ae1-bdfe-606573aa7af6'
               -> AND year = '2024'
               -> AND month = '11'
               -> AND lineitem_productcode = 'AmazonEC2'
               -> AND product_productfamily LIKE '%Compute Instance%'
               -> AND lineitem_resourceid != ''
               -> AND lineitem_usagestartdate = (
               ->     SELECT max(date(lv.lineitem_usagestartdate)) AS usage_start
               ->     FROM aws_line_items_daily AS lv
               ->     WHERE lineitem_resourceid = alid.lineitem_resourceid
               ->     AND year = '2024'
               ->     AND month = '11'
               ->     AND source = 'b52d9062-c53d-4ae1-bdfe-606573aa7af6'
               -> )
               -> GROUP BY
               ->     lineitem_resourceid,
               ->     product_instancetype,
               ->     resourcetags,
               ->     costcategory;
 resource_id | instance_type |  instance_name  |                                                                                                                                                                                 tags               >
-------------+---------------+-----------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
 i-444444442 | m5.large      | NULL            | {"CoM_RedHat_Rhel_varianT": "SAP", "com_REDHAT_rhel": "RHEL 7 ELS", "com_redhat_rhel_sla": "self-support", "com_redhat_rhel_usage": "disaster Recovery", "dashed-key-on-aws": "dashed-value", "envi>
 i-55555558  | m5.large      | NULL            | {"Map": "d2", "Mapping": "d1", "dashed-key-on-aws": "dashed-value", "version": "master"}                                                                                                           >
 i-444444445 | m5.large      | NULL            | {"CoM_RedHat_Rhel_varianT": "SAP", "com_REDHAT_rhel": "RHEL 8 ELS", "com_redhat_rhel_sla": "self-support", "com_redhat_rhel_usage": "disaster Recovery", "dashed-key-on-aws": "dashed-value", "envi>
 i-55555555  | m5.large      | instance-name-1 | {"Map": "a2", "Mapping": "a1", "Name": "instance-name-1", "dashed-key-on-aws": "dashed-value", "version": "prod"}                                                                                  >
 i-55555556  | m5.large      | instance-name-2 | {"CoM_RedHat_Rhel_varianT": "Workstation", "Map": "b2", "Mapping": "b1", "Name": "instance-name-2", "com_REDHAT_rhel": "8", "com_REDHAT_rhel_addon": "ELS", "com_redhat_rhel_conversion": "True", ">
 i-444444443 | m5.large      | instance-name-5 | {"CoM_RedHat_Rhel_varianT": "hPC", "Name": "instance-name-5", "com_REDHAT_rhel": "7", "com_redhat_rhel_sla": "self-support", "com_redhat_rhel_usage": "disaster Recovery", "dashed-key-on-aws": "da>
 i-444444446 | m5.large      | instance name 4 | {"CoM_RedHat_Rhel_varianT": "hPC", "Name": "instance name 4", "com_REDHAT_rhel": "8", "com_REDHAT_rhel_addon": "ELS", "com_REDHAT_rhel_conversion": "True", "com_redhat_rhel_sla": "self-support", >
 i-55555557  | m5.large      | instance_name_3 | {"CoM_RedHat_Rhel_varianT": "Workstation", "Map": "c2", "Mapping": "c1", "Name": "instance_name_3", "com_REDHAT_addon": "ELS", "com_REDHAT_conversion": "True", "com_REDHAT_rhel": "7", "com_redhat>
(8 rows)
```
Great Job Lucas!